docs/index.md

```diff
@@ -29,7 +29,46 @@ resource "illumio-cloudsecure_aws_account" "example" {
 }
 ```
 
+## Authentication and Configuration
+
+Illumio CloudSecure provides several methods for configuring the Terraform provider:
+
+1. Parameters in the provider configuration
+2. Environment variables
+
+Illumio CloudSecure follows the industry-standard protocol for authorization using OAuth 2.0. To generate the necessary client_id and client_secret to use the CloudSecure Terraform provider, you will need to create a Service Account from the [console](https://console.illum.io/#/serviceAccounts) and generate a new secret.
+
+| :warning: WARNING:                                                                                                                                                        |
+| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Hard-coded credentials are not recommended in any Terraform configuration and risks secret leakage should this file ever be committed to a public version control system. |
+
+Credentials can be provided by adding a client_id, client_secret, and optionally token, to the illumio-cloudsecure provider block.
+
+Usage:
+
+```terraform
+provider "illumio-cloudsecure" {
+  client_id = "my-access-id"
+  client_secret = "my-secret-id"
+}
+```
+
+### Environment Variables
+
+Credentials can be provided by using the `ILLUMIO_CLOUDSECURE_CLIENT_ID` and `ILLUMIO_CLOUDSECURE_CLIENT_SECRET` environment variables.
+
+```terraform
+provider "illumio-cloudsecure" {}
+```
+
+```
+% export ILLUMIO_CLOUDSECURE_CLIENT_ID="my-access-id"
+% export ILLUMIO_CLOUDSECURE_CLIENT_SECRET="my-secret-id"
```

Comment:
```suggestion
% export TF_VAR_illumio_cloudsecure_client_id="my-client-id"
% export TF_VAR_illumio_cloudsecure_client_secret="my-client-secret"
```

Alternatively, these variables can be set as [command-line arguments](https://developer.hashicorp.com/terraform/language/values/variables#variables-on-the-command-line), or in a [`.tfvars` file](https://developer.hashicorp.com/terraform/language/values/variables#variable-definitions-tfvars-files).